bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/regexp/RegExpFactory.java

```diff
@@ -124,7 +124,8 @@ public static void parseInsertion(String regExpStr) {
             treeBuilder.parseInsertion();
         } catch (BallerinaException e) {
             throw ErrorCreator.createError(BallerinaErrorReasons.REG_EXP_PARSING_ERROR,
-                    StringUtils.fromString("Invalid insertion in regular expression: " + e.getMessage()));
+                    StringUtils.fromString(e.getMessage() + " in insertion substring '" 
```

Comment:
Pass the original pattern here and use it in the error msg. 
What was the reason for wrapping the inclusion inside a non-capturing group?

Reply:
We do not have the original full pattern. We only have the inclusion part.

Reply:
AFIK, an inclusion should be in a place where it is a `ReDisjunction`. In a non-capturing group there is a `ReDisjunction` part after the colon. That's why we have modeled it in that way.